### ISSUE TITLE
[Feature] Turn off Stream Deck on system shutdown

### ISSUE THREAD

**Issue by Nillinn** [StreamController/StreamController#298]
Turn off devices on system shutdown / hibernate / etc.

When shutting down the computer, the Stream Deck remains on.  Doesn't happen on Windows.
The Stream Deck app for Windows manages to properly turn off every stream deck device before a system shutdown.
Stream Controller could use this feature if possible.

Could also use a terminal command for shutting down every Elgato device, so the user itself can implement it on a custom shutdown script

**Comment by oparada1988** [comment]
Can you share more details on this issue? Like do you see the elgato logo when turning off your PC?

**Comment by Nillinn** [comment]
@oparada1988 The Elgato logo doesn't show up like it does on Windows. The deck just stays on with its custom buttons and all, as if the PC was still up and running. Now I'm not sure if that's its normal behavior or if it only happens to me. 

I'm using Arch Linux. Maybe it doesn't turn off peripherals on shutdown by default, such as RGB mouse and keyboard? Since I'm not using any RGB peripherals I can't really tell if I'm missing some system-level setting

**Comment by oparada1988** [comment]
It shouldnt do that, it should turn off completely. I wonder if your BIOS has a setting where it will conitnue to provide power to a device attached to a USB 3 port. Id start there, is your streamdeck connected to a USB 3 port? Have you tried a USB 2 port to see if it does the same thing?

**Comment by Nillinn** [comment]
Thank you, it was indeed a BIOS setting. "USB power delivery in Soft Off state" (Asus motherboard). Comes enabled by default, I had to disable it. Some Windows apps seem to bypass this setting, so I never had trouble with any devices remaining on, there.

But that's solved the issue. Thanks.

**Comment by SchultzenDK** [comment]
Sorry, but can this issue be reopened?

I need the "USB power delivery" BIOS setting on, because I have a dock for my wireless mouse, and if I disable that, my mouse won't charge when my PC is off.

Power off works just fine when shutting down from Windows. It shows the Elgato logo and then powers off. But with StreamController on Linux, it does nothing and stays on.  
Screen lock works fine though. When I lock my computer, the deck goes black.

The deck is the MK2 with scissor keys, and I'm running the flatpak version on CachyOS (Arch)

**Comment by SchultzenDK** [comment]
I switched to OpenDeck which also has the same issue, but I found a way around it, which probably also works with StreamController.

TL;DR: Write a script that kills the StreamController (or OpenDeck) process and sleeps the USB, and set up a service that calls that script on shutdown.

Note: This was written with AI assistance, as I'm not a Linux wiz.

Run `lsusb` and save your vendor ID. We only care about the 4 digets I've marked as X'es: Bus 001 Device 001: ID XXXX:0000 Elgato Systems GmbH Stream Deck MK.2

There's a good chance it'll be 0fd9, unless your thingy is not an Elgato.

Create a script `/usr/local/bin/streamdeck-off.sh` or whatever you feel like calling it.

```bash
#!/bin/bash
# Replace XXXX with the vendor ID you found earlier
DEVICES=$(find /sys/bus/usb/devices/*/idVendor -exec grep -l "XXXX" {} +)

# Kill OpenDeck (can probably just replace with StreamController, otherwise check `htop` or `top`)
pkill opendeck

for dev_path in $DEVICES; do
    # Get the parent directory (the actual USB device folder)
    PARENT_DIR=$(dirname "$dev_path")
    
    # Force the device into "auto" suspend mode
    echo "0" > "$PARENT_DIR/power/autosuspend_delay_ms" 2>/dev/null
    echo "auto" > "$PARENT_DIR/power/control" 2>/dev/null
done
```

Make sure the script can be executed, and that it's owned by root, because the suspend mode requires sudo or root
```
sudo chown root:root /usr/local/bin/streamdeck-off.sh
sudo chmod +x /usr/local/bin/streamdeck-off.sh
```

Create service `/etc/systemd/system/streamdeck-off.service`
```yaml
[Unit]
Description=Turn off Stream Deck on shutdown
# We want this to run before the system actually starts killing processes
DefaultDependencies=no
Before=shutdown.target reboot.target halt.target

[Service]
Type=oneshot
# Run as root so it has permission to write to /sys/
User=root
Group=root
ExecStart=/usr/bin/bash /usr/local/bin/streamdeck-off.sh
# Ensures systemd doesn't think it failed if pkill doesn't find opendeck
SuccessExitStatus=0 1

[Install]
WantedBy=shutdown.target reboot.target halt.target
```

Enable the service `sudo systemctl enable streamdeck-off.service`

Again: The script and service was written with AI assistance, so it may not be the best or most optimal way to do it, but I've read through and understood what it does
